swingset/src/main/java/com/nqadmin/swingset/SSDataNavigator.java

```diff
@@ -1350,6 +1349,147 @@ public void setRowSet(final RowSet _rowSet) {
 //		setRowSet(_rowSet);
 //	}
 
+	private void setRowModified(boolean isDirty) {
+		isRowModified = isDirty;
+		if (!isDirty) {
+			errorComponents.clear();
+		}
+	}
+
+	private void updb(JButton b, boolean flag) {
```

Comment:
What is updb doing?

Reply:
updateButtonEnableState. It's a minimal overhead way to set a buttons state. It's used a lot. (but it probably doesn't matter overall performance); it is a one liner.